integration/common/osio.go

```diff
@@ -41,6 +42,26 @@ func StartServer(port int, handler func(w http.ResponseWriter, r *http.Request))
 	return
 }
 
+// WaitTillReady calls url every seconds for 10 times.
+func WaitTillReady(url string) error {
+	pingReq, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return err
+	}
+	for i := 0; i < 10; i++ {
+		res, err := http.DefaultClient.Do(pingReq)
+		if err == nil {
+			if res != nil {
+				if res.StatusCode == http.StatusOK {
```

Comment:
Here it’s not critical but it’s a good practice  always to read res.Body and then close it. Otherwise you will leak File Descriptors.
Here is an example from Auth service:
https://github.com/fabric8-services/fabric8-auth/blob/master/rest/rest.go#L113
https://github.com/fabric8-services/fabric8-auth/blob/81bf87a6bdae7528a88041f0493555a9fd45a847/auth/authz.go#L45